python_modules/dagster/dagster/_daemon/auto_run_reexecution/auto_run_reexecution.py

```diff
@@ -22,68 +24,54 @@
 DEFAULT_REEXECUTION_POLICY = ReexecutionStrategy.FROM_FAILURE
 
 
-def filter_runs_to_should_retry(
-    runs: Sequence[DagsterRun], instance: DagsterInstance, default_max_retries: int
-) -> Iterator[Tuple[DagsterRun, int]]:
-    """Return only runs that should retry along with their retry number (1st retry, 2nd, etc.)."""
-
-    def get_retry_number(run: DagsterRun) -> Optional[int]:
-        if run.status != DagsterRunStatus.FAILURE:
-            return None
-
-        raw_max_retries_tag = run.tags.get(MAX_RETRIES_TAG)
-        if raw_max_retries_tag is None:
-            max_retries = default_max_retries
-        else:
-            try:
-                max_retries = int(raw_max_retries_tag)
-            except ValueError:
-                instance.report_engine_event(
-                    f"Error parsing int from tag {MAX_RETRIES_TAG}, won't retry the run.", run
-                )
-                return None
-
-        if max_retries == 0:
-            return None
-
-        # TODO: group these to reduce db calls
-        run_group = instance.get_run_group(run.run_id)
-
-        if run_group:
-            _, run_group_iter = run_group
-            run_group_list = list(run_group_iter)
-
-            # Has the parent run already been retried the maximum number of times? (Group includes the parent)
-            if len(run_group_list) >= max_retries + 1:
-                return None
-
-            # Does this run already have a child run?
-            if any([run.run_id == run_.parent_run_id for run_ in run_group_list]):
-                return None
-            return len(run_group_list)
+def get_retry_number_if_should_retry(
+    run: DagsterRun,
+    instance: DagsterInstance,
+) -> Optional[int]:
+    """If the run should retry, returns the retry number. Otherwise, returns None."""
+    will_retry_tag_value = run.tags.get(WILL_RETRY_TAG)
+    if will_retry_tag_value is None:
```

Comment:
is there a test that would fail if we accidentally made this `if not will_retry_tag_value` instead of `if will_retry_tag_value is None` - I guess that wouldn't be the end of the world since in theory the value should be the same even if it needs to re-compute it here

Reply:
The values `will_retry_tag_value` could be are the string `"true"`, `"false"`, or `None` so i think if the behavior were changed to `if not will_retry_tag_value` the behavior would stay the same, right?

Reply:
that is a good point